### PR TITLE
Add metric tiers configuration

### PR DESCRIPTION
**Pull request by alexkorotkikh**
Cache tiers configuration and limits in KairosDB

### CODE REVIEW

File: src/main/java/org/kairosdb/core/tiers/MetricTiersConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    public boolean isMetricActiveForQuery(final String metricName) {
+        if (queryMaxCheckTier <= 0 || queryMaxCheckTier >= 3) {
+            return true;
+        }
+
+        final String[] split = metricName.split("\\.");
+        if (split.length != 3 || !split[0].equals("zmon") || !split[1].equals("check")) {
+            return false;
```
Comment (pitr) [inline src/main/java/org/kairosdb/core/tiers/MetricTiersConfiguration.java:25]: this means internal kairosdb metrics can't be fetched if tier throttling is in effect. I think this and next return should be true.

File: src/main/java/org/kairosdb/core/tiers/MetricTiersConfigurationUpdateJob.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    public static final Logger logger = LoggerFactory.getLogger(MetricTiersConfigurationUpdateJob.class);
+
+    private MetricTiersConfiguration config;
+    private ObjectMapper objectMapper;
+    private Executor executor;
+    private AccessToken accessToken;
+    private String schedule;
+    private String hostname;
+
+    MetricTiersConfigurationUpdateJob() {
```
Comment (pitr) [inline src/main/java/org/kairosdb/core/tiers/MetricTiersConfigurationUpdateJob.java:33]: is the default constructor needed

File: src/main/java/org/kairosdb/core/tiers/MetricTiersConfiguration.java (hunk 1 above)
Comment (alexkorotkikh) [reply src/main/java/org/kairosdb/core/tiers/MetricTiersConfiguration.java:25]: I believe that kairosdb internal metrics are the last thing we have to care about if under unexpected load

File: src/main/java/org/kairosdb/core/tiers/MetricTiersConfigurationUpdateJob.java (hunk 2 above)
Comment (alexkorotkikh) [reply src/main/java/org/kairosdb/core/tiers/MetricTiersConfigurationUpdateJob.java:33]: it's required by Mockito

File: src/main/java/org/kairosdb/core/tiers/MetricTiersConfiguration.java (hunk 1 above)
Comment (pitr) [reply src/main/java/org/kairosdb/core/tiers/MetricTiersConfiguration.java:25]: doesn't that mean we would lose visibility into our internals then?

Comment (alexkorotkikh) [reply src/main/java/org/kairosdb/core/tiers/MetricTiersConfiguration.java:25]: I believe metrics on our internals are not considered as critical

**Review by pitr — APPROVED** [review]

**Comment by pitr** [comment]
👍

**Comment by alexkorotkikh** [comment]
👍

**Pull request merged**